versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -748,16 +785,35 @@ public PaginationIterator<Diff> getDiffs(Ref from, Ref to, String pagingToken)
       emptyOrNotFound(to, null);
     }
 
-    PagingToken token = pagingToken != null ? fromString(pagingToken) : null;
-    StoreKey start = token != null ? keyFromString(token.token().toStringUtf8()) : null;
-    StoreKey end = null;
-
     CommitLogic commitLogic = commitLogic(persist);
     PagedResult<DiffEntry, StoreKey> diffIter =
-        commitLogic.diff(diffQuery(token, fromCommit, toCommit, start, end, true));
+        commitLogic.diff(
+            diffQuery(
+                keyRanges.pagingTokenObj(),
+                fromCommit,
+                toCommit,
+                keyRanges.beginStoreKey(),
+                keyRanges.endStoreKey(),
+                true));
 
     ContentMapping contentMapping = new ContentMapping(persist);
 
+    Predicate<DiffEntry> keyPred =
+        contentKeyPredicate != null
+            ? d -> {
+              ContentKey key = storeKeyToKey(d.key());
+              return key != null && contentKeyPredicate.test(key);
+            }
+            : x -> true;
+
+    Predicate<DiffEntry> stopPredicate;
+    if (prefixKey != null) {
+      StoreKey prefix = keyToStoreKeyNoVariant(prefixKey);
+      stopPredicate = d -> !d.key().startsWithElementsOrParts(prefix);
```

Comment:
`startsWithElementsOrParts` seems to contradict the doc text in `ApiDoc`... which way do we want it to work? :)

Reply:
doc's wrong, fixed